nodejs/week2/app.js

```diff
@@ -0,0 +1,67 @@
+
+const { json } = require("express");
+const express = require("express");
+const app = express();
+const data = require("./document.json");
+const port = process.env.PORT || 3000;
+app.use(express.json())
+app.get('/', (req, res) => {
+    res.send('This is a search engine')
+})
+/* 
+app.get('/search', (req, res) => {
+ const searchWord = req.query.q
+ const listItem = [];
+  data.forEach(element => {
+    let itemFound = false;
+        Object.keys(element).forEach(elementKey => {
+            if(new String(element[elementKey]).includes(searchWord) && !itemFound){
+               listItem.push(element)
+               itemFound = true;
+            }
+        
+        });
+    });
+    if(listItem.length === 0){
+    res.status(404).json("Search Word is not availbel in Data")
+    }
+    else{
+        res.json(listItem)
+    }
+}) */
+
+app.get('/search', (req, res) => {
+    const searchWord = req.query.q
+    const listItem = data.filter(e => Object.values(e).some(value => String(value).includes(searchWord)))
+    if (listItem.length === 0) {
+        res.status(404).json("Search Word is not availbel in Data")
```

Comment:
According to exercise, if no query params are provided, then it should return all the records in json file.

Reply:
Yahh. you are right.. I will update code.